src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,102 @@
+package duke;
+
+import duke.command.Command;
+import duke.exception.DukeException;
+import duke.util.Parser;
+import duke.util.Storage;
+import duke.util.TaskList;
+import duke.util.Ui;
+
+/**
+ * Represents a ChatBot.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+    private boolean isExit = false;
+    /**
+     * Constructs a new Duke instance to run.
+     * Reads from file to set up chatbot.
+     *
+     * @param file Name of file to save data and read from.
+     * @param fileParent Parent directory of file.
+     * @param name Name of chatbot.
+     * @param logo Logo of chatbot.
+     */
+    public Duke(String file, String fileParent, String name, String logo) {
```

Comment:
Instead of taking in the `file` and the `fileParent`, you could derive the parent path from the full file path. 

For example
```java
import java.io.File;
import java.nio.file.Path;
import java.nio.file.Paths;
Path path = Paths.get(filepath);
File directory = path.getParent().toFile();
directory.mkdirs();
this.file = path.toFile();
this.file.createNewFile();
```
The above code would get the Path of the directory, create it if it hasn't been created, and then create the file if it hasn't been created also.